metrics/options/options.go

```diff
@@ -36,14 +38,20 @@ type HeapsterRunOptions struct {
 }
 
 func NewHeapsterRunOptions() *HeapsterRunOptions {
-	return &HeapsterRunOptions{}
+	opt := genericoptions.NewServerRunOptions()
+	return &HeapsterRunOptions{
+		ServerRunOptions: opt,
+	}
 }
 
 func (h *HeapsterRunOptions) AddFlags(fs *pflag.FlagSet) {
+	h.ServerRunOptions.AddUniversalFlags(pflag.CommandLine)
+
 	fs.Var(&h.Sources, "source", "source(s) to watch")
 	fs.Var(&h.Sinks, "sink", "external sink(s) that receive data")
 	fs.DurationVar(&h.MetricResolution, "metric_resolution", 60*time.Second, "The resolution at which heapster will retain metrics.")
-	fs.IntVar(&h.Port, "port", 8082, "port to listen to")
+	// TODO: Revise this flag before Heapster v1.3 and Kubernetes v1.5
+	fs.IntVar(&h.Port, "heapster-port", 8082, "port used by the Heapster-specific APIs")
```

Comment:
We are not sure how to handle auth, so intially we would like to have two ports: for the native heapster api and metrics api server via api server. Eventually native api should be server on the same port as an addtional handlers in genericapiserver library.

I agree, but this is because there is already deprecated flag in genericapiserver with the same name and we have a conflict. Eventually (before releasing next Heapster version) we want to either remove the flag from genericapiserver or move it to apiserver and then revert this change here.